src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,90 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final double MAX_CAPACITY = 0.75;
+    private static final int INDEX_OF_CAPACITY = 2;
+    private int size;
+    private Node<K, V>[] elements;
+
+    {
+        elements = (Node<K, V>[]) new Node[DEFAULT_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
+        resizeIfNeed();
+        int currentCapacity = elements.length;
+
+        int hash = (key == null) ? 0 : Math.abs(key.hashCode() % elements.length);
 
+        Node<K, V> currentNode = elements[hash];
+
+        while (currentNode != null) {
+            if (currentNode.key == key || (key != null && key.equals(currentNode.key))) {
+                currentNode.value = value;
+                return;
+            }
+            currentNode = currentNode.next;
+        }
+        Node<K, V> newNode = new Node<>(key, value, elements[hash]);
+        elements[hash] = newNode;
+        size++;
     }
 
     @Override
     public V getValue(K key) {
+        for (Node<K, V> currentNode : elements) {
+            while (currentNode != null) {
+                if (currentNode.key == key || (key != null && key.equals(currentNode.key))) {
+                    return currentNode.value;
+                }
+                currentNode = currentNode.next;
+            }
+        }
```

Comment:
The `getValue` method should not iterate over all elements. Instead, it should calculate the hash for the given key and only traverse the linked list at that specific index. This is a critical issue as it affects the performance and correctness of the hash map.